### ISSUE TITLE
Mypy does not accept Literal StrEnum types

### ISSUE THREAD

**Issue by ericmburgess** [irgeek/StrEnum#18]
Under `mypy 0.991`, it doesn't appear that we can use `Literal` types whose values are `StrEnum` elements. With the following sample code:

```python
from enum import Enum
from typing import Literal, TypedDict

from strenum import StrEnum


class Entree(str, Enum):
    spam = "SPAM"
    eggs = "EGGS"


class Side(StrEnum):
    bacon = "BACON"
    grits = "GRITS"


class Breakfast(TypedDict):
    entree: Entree
    side: Side


class EggBreakfast(TypedDict):
    entree: Literal[Entree.eggs]
    side: Side


class BaconBreakfast(TypedDict):
    entree: Entree
    side: Literal[Side.bacon]
```

The `mypy` output is:

```
type-error.py:28: error: Variable "type-error.Side.bacon" is not valid as a type  [valid-type]
type-error.py:28: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
type-error.py:28: error: Parameter 1 of Literal[...] is invalid  [valid-type]
Found 2 errors in 1 file (checked 1 source file)
```

Note that no error is thrown for `Literal[Entree.eggs]`, which is defined on a `(str, Enum)` value rather than a `StrEnum` value.

I don't know whether this can be resolved from outside `mypy`, but I wanted to make the maintainers here aware if they're not, and find out if there's a better way to do this that I haven't come up with yet. In particular, `Literal["BACON"]` does remove the error, but is clearly not ideal.

**Comment by irgeek** [comment]
Hi, and thanks for taking the time to report this.

I've just created PR #19 which adds stub files with type hints. When I tested locally, `mypy` was able to check your example without finding any errors. I don't use `mypy` much, though, so I'd appreciate other eyes taking a look before I merge them.

Can you check if this fixes the error for you, please?

**Comment by irgeek** [comment]
Fixed in `v0.4.10`.
